### PR TITLE
Enhancement: Enable and configure header_comment fixer

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] enables and configures the `header_comment` fixer
* [x] runs `make cs`

Follows #157.

💁‍♂️ For reference, see https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.17.2/doc/rules/comment/header_comment.rst.

### CODE REVIEW

File: .php_cs.dist [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 declare(strict_types=1);
 
+$header = <<<TXT
+Copyright (c) 2011 François Zaninotto and contributors
```
Comment (localheinz) [inline .php_cs.dist:6]: For reference, see https://github.com/FakerPHP/Faker/pull/154/files#r546353196.

**Comment by krsriq** [comment]
That's a big one.. +5439 LOC, are we sure we need that? What's the benefit of having the copyright message in every file?

**Comment by localheinz** [comment]
@krsriq 

> That's a big one.. +5439 LOC, are we sure we need that? What's the benefit of having the copyright message in every file?

🤷‍♂️ It is a practice I have observed in every major PHP project.

Also see https://softwareengineering.stackexchange.com/q/125836/209095.

**Comment by mfn** [comment]
Will this apply to files which were not in the previous repo, too? Does that even make sense?

For example, the header is _also_ added to `.php_cs.dist` which however did not exist hours before, so attributing copyright to the former fork owner seems strange?

If we're happy with an all-compassing license, can't we just have a single one in the root? Isn't it accepted it applies to the whole repo?

Spoiler: it _is_ already there and in fact contains a few other copyright holder referenced that are not mentioned in this one here.

**Comment by krsriq** [comment]
@localheinz 
> It is a practice I have observed in every major PHP project.

There are also exceptions - for example [Guzzle](https://github.com/guzzle/guzzle). [Laravel](https://github.com/laravel/framework) also has a different solution: one LICENSE.md in each directory.

**Comment by localheinz** [comment]
@mfn 

> Will this apply to files which were not in the previous repo, too? 

Yes.

>Does that even make sense?

Does it not?

>For example, the header is also added to .php_cs.dist which however did not exist 24 hours before, so attributing copyright to the former fork owner seems strange?

Does it?

If you look at the source code of `friendsofphp/php-cs-fixer`, you can find @fabpot mentioned in every file-level DocBlock, although he hasn't contributed to `friendsofphp/php-cs-fixer` in years.

>If we're happy with an all-compassing license, can't we just have a single one in the root? Isn't it accepted it applies to the whole repo?

See https://softwareengineering.stackexchange.com/q/125836/209095.

> Spoiler: it is already there and in fact contains a few other copyright holder referenced that are not mentioned in this one here.

That's why I propose the phrase _and contributors_.

**Comment by localheinz** [comment]
@krsriq 

> There are also exceptions - for example [Guzzle](https://github.com/guzzle/guzzle). [Laravel](https://github.com/laravel/framework) also has a different solution: one LICENSE.md in each directory.

Interesting. I'm proposing a different solution here.

We can

- discuss it
- reject it
- accept it

I'm fine with any of these.

**Comment by krsriq** [comment]
@localheinz 

> discuss it

Yes, that's what we're doing :)

- Having a copyright header comment in every file is not required for the license to be binding (the official [GitHub guide on the MIT License](https://choosealicense.com/licenses/mit/) says "Create a text file (typically named LICENSE or LICENSE.txt) in the root of your source code and copy the text of the license into the file."). 
- It may be the case that the majority of OSS (PHP) projects do it that way, but there are some major projects that don't.
- It increases the overall project file size.
- It is an additional thing that has to be considered for each file creation.

In short, it adds overhead without being necessary, therefore I suggest to reject this one.

**Comment by localheinz** [comment]
@krsriq 


> Having a copyright header comment in every file is not required for the license to be binding (the official [GitHub guide on the MIT](https://choosealicense.com/licenses/mit/) License says "Create a text file (typically named LICENSE or LICENSE.txt) in the root of your source code and copy the text of the license into the file." - the [Gnu](https://choosealicense.com/licenses/agpl-3.0/) and [Apache](https://choosealicense.com/licenses/apache-2.0/) license guides mention the optional step of adding a notice to each file).

Yes, that's what the answers to the [linked question on StackExchange](https://softwareengineering.stackexchange.com/q/125836/209095) suggest, too.

> It may be the case that the majority of OSS (PHP) projects do it that way, but there are some major projects that don't.

Yes, that's what the answers to the [linked question on StackExchange](https://softwareengineering.stackexchange.com/q/125836/209095) suggest, too.

> It increases the overall project file size.

It does, yes.

> It is an additional thing that has to be considered for each file creation.

It does not, that's what why I suggest to configure the `header_comment` fixer here. A developer needs to run

```
$ make cs
```

to apply these changes, that is all.

> In short, it adds overhead without being necessary, therefore I suggest to reject this one.

**Comment by Nyholm** [comment]
I vote no. 

It does not bring any clear benefits and I also would like to bring up the ship of Theseus. 

Guzzle is not using the file header and we don't plan to add it =)
